fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java

```diff
@@ -86,39 +93,26 @@ public class SmsMessage extends AbstractPersistableCustom {
 
     public static SmsMessage pendingSms(final String externalId, final Group group, final Client client, final Staff staff,
             final String message, final String mobileNo, final SmsCampaign smsCampaign, final boolean isNotification) {
-        return new SmsMessage(externalId, group, client, staff, SmsMessageStatusType.PENDING, message, mobileNo, smsCampaign,
-                isNotification);
+        return new SmsMessage().setExternalId(externalId).setGroup(group).setClient(client).setStaff(staff)
+                .setStatusType(SmsMessageStatusType.PENDING.getValue()).setMessage(message).setMobileNo(mobileNo)
+                .setSmsCampaign(smsCampaign).setNotification(isNotification);
     }
 
     public static SmsMessage sentSms(final String externalId, final Group group, final Client client, final Staff staff,
             final String message, final String mobileNo, final SmsCampaign smsCampaign, final boolean isNotification) {
-        return new SmsMessage(externalId, group, client, staff, SmsMessageStatusType.WAITING_FOR_DELIVERY_REPORT, message, mobileNo,
-                smsCampaign, isNotification);
+        return new SmsMessage().setExternalId(externalId).setGroup(group).setClient(client).setStaff(staff)
```

Comment:
Missing: `this.submittedOnDate = DateUtils.getBusinessLocalDate();`

Reply:
done